Exposing command line flags currently supported in clm preprocess in the snakefile.

anushka255 opened the pull request:
- The `no_neutralise` flag is exposed as `neutralise` in the config file, and the logic is worked out accordingly in the Snakefile.

vineetbansal commented on workflow/config/config.yaml:
```diff
@@ -46,6 +46,18 @@ enum_factors:
+  min_heavy_atoms: 3
+  # Defines the set of elements required for a molecule to be considered valid.
+  # Any SMILES containing elements outside this set will be considered invalid and excluded from the training set.
+  valid_atoms: ["Br", "C", "Cl", "F", "H", "I", "N", "O", "P", "S"]
```
Does this syntax work for lists in yaml? Perhaps it does, but better to make sure that its not being passed in as one long string or something (which would mean that C gets included implicitly anytime Cl gets included etc). Also the quotes may not be needed, but if they're used, can we ensure that they're not (unnecessarily) being passed to the function as escaped quotes?

anushka255 replied:
Yes, it does work. This is how it's passed on as: `--valid-atoms Br C Cl F H I N O P S ` . Because the `valid-atoms` flag has a `nargs` argument set to '+', the above string is interpreted as a list separated by space.  Also, the quotes aren' t affecting anything down the line, but I will remove those as they seem obsolete.

vineetbansal commented on workflow/Snakefile_data:
```diff
@@ -64,7 +64,11 @@ rule preprocess:
+        '--min-heavy-atoms {config[preprocess][min_heavy_atoms]} '
+        '--valid-atoms {config[preprocess][valid_atoms]} '
         '--max-input-smiles {config[max_input_smiles]} '
+        f'{"--no_neutralise " if not config["preprocess"]["neutralise"] else ""} '
```
It looks to me like we're unnecessarily converting from neutralize->no_neutralize->neutralize in the code. Maybe we can go with a neutralize flag throughout.

anushka255 replied:
I am only exposing the variable as `neutralize` in the config file because the double negative that the `no_neutralize` flag might call for tends to be confusing. If we were to convert the `no_neutralize` flag  to `neutralize` , we will have to change its default value from `False` to `True`  which could lead to more confusion.

vineetbansal commented:
@anushka255 - for the commits after the approval - I'm not a great fan of parameterization if its only used for a couple of combinations, and particularly if the number of parameters in each combination is many, as is the case here. Its not immediately clear whats different between the parameterized versions unless one goes back and forth between the body and the decorator, and one is forced to reason about all combinations even when just trying to debug a single one.

I hate to do this - but can we not just have two versions of the test function with comments if necessary?

If you still feel strongly about doing this, we can discuss..

anushka255 commented:
> @anushka255 - for the commits after the approval - I'm not a great fan of parameterization if its only used for a couple of combinations, and particularly if the number of parameters in each combination is many, as is the case here. Its not immediately clear whats different between the parameterized versions unless one goes back and forth between the body and the decorator, and one is forced to reason about all combinations even when just trying to debug a single one.
> 
> I hate to do this - but can we not just have two versions of the test function with comments if necessary?
> 
> If you still feel strongly about doing this, we can discuss..

You're right, the reduction in lines from parametrization isn't worth the trade-off in readability. I have removed the parameterization entirely and split up the test case into two.
